pcsx2/USB/usb-pad/usb-mascon.cpp

```diff
@@ -0,0 +1,585 @@
+// SPDX-FileCopyrightText: 2002-2024 PCSX2 Dev Team
+// SPDX-License-Identifier: GPL-3.0+
+
+#include "usb-mascon.h"
+
+#include "common/Console.h"
+#include "Host.h"
+#include "IconsPromptFont.h"
+#include "Input/InputManager.h"
+#include "StateWrapper.h"
+#include "USB/deviceproxy.h"
+#include "USB/USB.h"
+#include "USB/qemu-usb/USBinternal.h"
+#include "USB/qemu-usb/desc.h"
+
+namespace usb_pad
+{
+	const char* MasconDevice::Name() const
+	{
+		return TRANSLATE_NOOP("USB", "Densha Controller");
+	}
+
+	const char* MasconDevice::TypeName() const
+	{
+		return "DenshaCon";
+	}
+
+	std::span<const char*> MasconDevice::SubTypes() const
+	{
+		static const char* subtypes[] = {
+			TRANSLATE_NOOP("USB", "Type 2"),
+			TRANSLATE_NOOP("USB", "Shinkansen"),
+			TRANSLATE_NOOP("USB", "Ryojouhen"),
+		};
+		return subtypes;
+	}
+
+	enum MasconControlID
+	{
+		CID_MC_POWER,
+		CID_MC_BRAKE,
+		CID_MC_UP,
+		CID_MC_RIGHT,
+		CID_MC_DOWN,
+		CID_MC_LEFT,
+
+		// TCPP20009 sends the buttons in this order in the relevant byte
+		CID_MC_B,
+		CID_MC_A,
+		CID_MC_C,
+		CID_MC_D,
+		CID_MC_SELECT,
+		CID_MC_START,
+
+		// Ryojouhen controller has 7 buttons, map L/R onto existing indexes.
+		CID_MC_CAMERA,
+		CID_MC_L = CID_MC_C,
+		CID_MC_R = CID_MC_D,
+
+		BUTTONS_OFFSET = CID_MC_B,
+	};
+
+	std::span<const InputBindingInfo> MasconDevice::Bindings(u32 subtype) const
+	{
+		switch (subtype)
+		{
+			case MT_TYPE2:
+			case MT_SHINKANSEN:
+			{
+				static constexpr const InputBindingInfo bindings[] = {
+					{"Power", TRANSLATE_NOOP("Pad", "Power"), ICON_PF_LEFT_ANALOG_DOWN, InputBindingInfo::Type::Axis, CID_MC_POWER, GenericInputBinding::LeftStickDown},
```

Comment:
```suggestion
					{"Power", TRANSLATE_NOOP("USB", "Power"), ICON_PF_LEFT_ANALOG_DOWN, InputBindingInfo::Type::Axis, CID_MC_POWER, GenericInputBinding::LeftStickDown},
```
same for the entire file